contrib/CatchAddTests.cmake

```diff
@@ -48,6 +50,47 @@ endif()
 
 string(REPLACE "\n" ";" output "${output}")
 
+# Run test executable to get list of available reporters
+execute_process(
+  COMMAND ${TEST_EXECUTOR} "${TEST_EXECUTABLE}" ${spec} --list-reporters
+  OUTPUT_VARIABLE reporters_output
+  RESULT_VARIABLE reporters_result
+)
+if(${reporters_result} EQUAL 0)
+  message(WARNING
+    "Test executable '${TEST_EXECUTABLE}' contains no reporters!\n"
+  )
+elseif(${reporters_result} LESS 0)
+  message(FATAL_ERROR
+    "Error running test executable '${TEST_EXECUTABLE}':\n"
+    "  Result: ${reporters_result}\n"
+    "  Output: ${reporters_output}\n"
+  )
+endif()
+string(FIND "${reporters_output}" "${reporter}" reporter_is_valid)
+if(reporter AND ${reporter_is_valid} EQUAL -1)
+  message(FATAL_ERROR
+    "\"${reporter}\" is not a valid reporter!\n"
+  )
+endif()
+
+# Prepare report
+set(report_extension ".txt")
+if(reporter)
+  set(reporter_arg "-r ${reporter}")
+  if(reporter MATCHES "junit|xml")
+    set(report_extension ".xml")
+  endif()
```

Comment:
I am not quite sold on this -- these two aren't the only Catch2-provided reporters that produce XML output, and in general it seems very limited (if potentially useful).

I think the ideal solution would be to let the user set a filename pattern, like `output-{date}-{testname}.json`, but that brings in _a lot_ more complexity.

I'll have to think about this a bit more.

Reply:
I could always make `OUTPUT_DIR` a multi value keyword, optional 2nd value would be the file extension, defaulting to none:
```
catch_discover_tests(foo REPORTER junit OUTPUT_DIR test-results xml)
```

Let me know what you think.

Reply:
Agreed with 4) being the best option.